File: Tutoriels/Cpp/Multithreading/z1_Cpp_Multithreading_Cookbook/src/manager/GConfig.cpp
//================================================
#include "GConfig.h"
//================================================
GConfig* GConfig::m_instance = 0;
//================================================
GConfig::GConfig() {

}
//================================================
GConfig::~GConfig() {
	
}
//================================================
GConfig* GConfig::Instance() {
	if(m_instance == 0) {
		m_instance = new GConfig;
	}
	return m_instance;
}
//================================================
void GConfig::setData(const string& key, const string& data) {
	m_data[key] = data;
}
//================================================
string GConfig::getData(const string& key) {
	return m_data[key];
}
//================================================

File: Tutoriels/PHP/Projets/php/class/GConfig.php
<?php   
    class GConfig {
        //===============================================
        private static $m_instance = null;
		private $m_dataMap = array();
        //===============================================
        private function __construct() {
        
        }
        //===============================================
        public static function Instance() {
            if(is_null(self::$m_instance)) {
                self::$m_instance = new GConfig();  
            }
            return self::$m_instance;
        }
        //===============================================
        public function setData($key, $data) {
            $this->m_dataMap[$key] = $data;
        }
        //===============================================
        public function getData($key) {
            return $this->m_dataMap[$key];
        }
        //===============================================
        public function showData() {
			foreach($this->m_dataMap as $key => $data) {
				echo $key . " = " . $data . "<br>";
			}
			echo "<br>";
        }
        //===============================================
    }
?>